### PR TITLE
Write successes with fileappendingreporter

### PR DESCRIPTION
**Pull request by BenPH**
## Description
Adding the option to write successes with FileAppendingReporter

## Checklist

- [x] **I am adding new functionality.**
  - [x] I have added or updated the tests for the new or changed functionality in `Tests/UnitTests`.
  - [x] I have added note(s) to `CHANGELOG.md` as necessary.

## Contributing

- [x] I have read and agree to the [Contributor Agreement](https://github.com/sassoftware/jsl-hamcrest/blob/master/ContributorAgreement.txt)

Signed-off-by: Ben Peachey Higdon <bpeacheyhigdon@gmail.com>

### CODE REVIEW

**Comment by BenPH** [comment]
Hi @JustinChilton Could you have a look at these changes? I wanted to be able to write out successes as well as failures with the FileAppendingReporter. Thanks

File: Source/Reporters/FileAppendingReporter.jsl [hunk 1]
```diff
@@ -15,8 +15,9 @@ Define Class(
 	
 	filepath = "";
 	
-	_init_ = Method({filepath},
+	_init_ = Method({filepath, write successes=0},
```
Comment (JustinChilton) [inline Source/Reporters/FileAppendingReporter.jsl]: Remove `=0` default argument. The default is already in the factory function below, which should always be used to create this reporter.

File: Source/Reporters/FileAppendingReporter.jsl [hunk 2]
```diff
@@ -51,6 +52,10 @@ Define Class(
 	
 	// Function: add success
 	add success = Method( {label, test expr, description, payload=Empty()},
+		if(this:write successes,
```
Comment (JustinChilton) [inline Source/Reporters/FileAppendingReporter.jsl]: Can you capitalize `if` to `If`?

File: Source/Reporters/FileAppendingReporter.jsl [hunk 3]
```diff
@@ -61,13 +66,14 @@ Define Class(
...
     
     Example:
     ---JSL---
     ut global reporter = ut file appending reporter( "$DESKTOP/report.txt" );
     ---------
 */
-ut file appending reporter = Function({file},
-	New Object( "UtFileAppendingReporter"( file ) );
+ut file appending reporter = Function({file, write successes=0},
+	New Object( "UtFileAppendingReporter"( file, write successes) );
```
Comment (JustinChilton) [inline Source/Reporters/FileAppendingReporter.jsl]: Add a space after `write successes`.

File: Source/Reporters/FileAppendingReporter.jsl [hunk 4]
```diff
@@ -51,6 +52,10 @@ Define Class(
 	
 	// Function: add success
 	add success = Method( {label, test expr, description, payload=Empty()},
+		if(this:write successes,
+			success = Eval Insert( "\!NSUCCESS: ^Name Expr(test expr)^ ^description^ (^label^)" );
```
Comment (JustinChilton) [inline Source/Reporters/FileAppendingReporter.jsl]: Should we be using something like the `shortDescription` above? This description can be very large in some cases (e.g. large matrices).

**Review by JustinChilton — CHANGES_REQUESTED** [review]
Looks good. Just a few small changes needed.

File: Source/Reporters/FileAppendingReporter.jsl (hunk 4 above)
Comment (BenPH) [reply Source/Reporters/FileAppendingReporter.jsl]: I think you're right - it would get messy. I was debating it because failures show the full description as well but successes are less important anyway. I made the change.

**Review by JustinChilton — APPROVED** [review]
